Web/stockdogweb/src/components/Trade/Transaction/Transaction.js

```diff
@@ -0,0 +1,92 @@
+import React, { Component } from 'react';
+import Modal from 'react-responsive-modal';
+import Select from 'react-select';
+import Button from '../../Button/Button';
+import './Transaction.css';
+
+const selectOptions = [
+	{value: 'buy', label: 'Buy'},
+	{value: 'sell', label: 'Sell'}
+]
+
+
+class Transaction extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+        	selectedOption: {value: 'buy', label: 'Buy'},
+        	buttonDisabled: true,
+        	amount: ""
+        }
+    }
+
+    handleSelectChange = (selectedOption) => {
+    	this.setState({selectedOption});
+    }
+
+    onChange = (event) => {
+    	var newState = this.state;
+
+    	// Strip all non-numeral characters
+    	var amount = event.target.value.replace(/[^0-9.]/g,"");
+    	// If 0, replace with empty
+    	amount = amount === "0" ? "" : amount;
+    	newState[event.target.id] = amount;
+
+    	// Check to enable Execute button
+    	if (amount !== "" 
+    		&& this.props.price * amount <= this.props.buyingPower) {
```

Comment:
Does this condition also work for the selling transaction?

Reply:
You're right, I completed neglected selling stuff lol